src/renderer/components/pages/ConfigPage.tsx

```diff
@@ -583,6 +636,31 @@ export class ConfigPage extends React.Component<ConfigPageProps, ConfigPageState
     updatePreferencesData({ excludedRandomLibraries });
   }
 
+  onRemoveAppPathOverride = (index: number): void => {
+    const newPaths = [...this.props.preferencesData.appPathOverrides];
+    newPaths.splice(index, 1);
+    console.log('SPLICED');
+    updatePreferencesData({ appPathOverrides: newPaths });
+  }
+
+  onNewAppPathOverride = (): void => {
+    const newPaths = [...this.props.preferencesData.appPathOverrides];
+    newPaths.push({path: '', override: ''});
+    updatePreferencesData({ appPathOverrides: newPaths });
+  }
+
+  onAppPathOverridePathChange = (index: number, newPath: string): void => {
+    const newPaths = [...this.props.preferencesData.appPathOverrides];
+    newPaths[index].path = newPath;
```

Comment:
This mutates the current prop. It should be copied before editing (``newPaths[index] = { ...newPaths[index], path: newPath };``). Same with ``onAppPathOverrideOverrideChange``